### DISCUSSION TITLE
Can kong's embeded nginx upgrade to latest version?

### DISCUSSION THREAD

**Discussion by Chenjp** [Kong/kong#9170] [Help] [answered]
Current we want to upgrade nginx standalone to 1.22.0 because of security issue.
How to?

**Comment by chronolaw** [comment, marked as answer]
It is a possible but hard work.

Kong depends on OpenResty and has many other C patches, and Lua libraries, If you want to upgrade to your own Nginx, you must deal with all of these.

I think `kong-build-tools` is a good start, but you still need more effort.

**Comment by Chenjp** [comment]
Without kong / nginx upgrade kits, we have to accept the fact that our Kong API service is disarming against external cyber security threats (Nginx CVEs).

Will you accept it as a feature request?
